packages/select/src/vaadin-select-base-mixin.js

```diff
@@ -259,8 +259,7 @@ export const SelectBaseMixin = (superClass) =>
         menuElement.addEventListener(
           'click',
           () => {
-            this.__userInteraction = true;
-            this.opened = false;
```

Comment:
Here is a bit more context. The issue emerged when I prevented validation on close as part of the dirty state work. Validation didn't get triggered by this line anymore and `should validate before change event on Enter` started to fail when running with the Lit version of the component. It turned out that the change event was fired synchronously while all validation was performed asynchronously (delayed to the next update due to value property change).